space-cli/cmd/upgrade.go

```diff
@@ -0,0 +1,256 @@
+		return err
+	}
+
+	// get all containers containing < space-cloud > in their name
+	args := filters.Arg("name", "space-cloud")
+	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{Filters: filters.NewArgs(args), All: true})
+	if err != nil {
+		logrus.Errorf("Unable to list containers - %s", err.Error())
+		return err
+	}
+
+	//parameters for gateway
+	var gatewayMounts []mount.Mount
+	var gatewayPorts nat.PortMap
+	var gatewayEnvs []string
+	var latestVersion string
+
+	//parameters for runner
+	var runnerEnvs []string
+
+	// Remove all container
+	for _, containerInfo := range containers {
+		containerInspect, err := cli.ContainerInspect(ctx, containerInfo.ID)
+		if err != nil {
+			logrus.Errorf("error getting service in docker unable to inspect container - %v", err)
+			return err
+		}
+
+		imageName := strings.Split(containerInfo.Image, ":")
+		switch imageName[0] {
+		case "spaceuptech/gateway":
+
+			gatewayEnvs = containerInspect.Config.Env
+
+			result := make(map[string]interface{})
+			if err := Get(http.MethodGet, "/v1/config/env", map[string]string{}, &result); err != nil {
+				return err
+			}
+
+			currentVersion := result["version"].(string)
+			latestVersion, err = getLatestVersion(currentVersion)
+			if err != nil {
+				return err
+			}
+			if currentVersion == latestVersion {
+				return fmt.Errorf("current verion (%s) is up to date", currentVersion)
+			}
+
+			gatewayMounts = containerInspect.HostConfig.Mounts
+
+			gatewayPorts = containerInspect.HostConfig.PortBindings
+
+			if err := cli.ContainerRemove(ctx, containerInfo.ID, types.ContainerRemoveOptions{Force: true}); err != nil {
+				logrus.Errorf("Unable to remove container %s - %s", containerInfo.ID, err.Error())
+				return err
+			}
+
+		case "spaceuptech/runner":
+			runnerEnvs = containerInspect.Config.Env
+
+			if err := cli.ContainerRemove(ctx, containerInfo.ID, types.ContainerRemoveOptions{Force: true}); err != nil {
+				logrus.Errorf("Unable to remove container %s - %s", containerInfo.ID, err.Error())
+				return err
+			}
+		}
+	}
+
+	containersToCreate := []struct {
+		dnsName        string
+		containerImage string
+		containerName  string
+		envs           []string
+		mount          []mount.Mount
+		exposedPorts   nat.PortSet
+		portMapping    nat.PortMap
+	}{
+		{
+			containerImage: fmt.Sprintf("%s:v%s", "spaceuptech/gateway", latestVersion),
+			containerName:  ContainerGateway,
+			dnsName:        "gateway.space-cloud.svc.cluster.local",
+			envs:           gatewayEnvs,
+			exposedPorts: nat.PortSet{
+				"4122": struct{}{},
+				"4126": struct{}{},
+			},
+			portMapping: gatewayPorts,
+			mount:       gatewayMounts,
+		},
+
+		{
+			// runner
+			containerImage: fmt.Sprintf("%s:v%s", "spaceuptech/runner", latestVersion),
+			containerName:  ContainerRunner,
+			dnsName:        "runner.space-cloud.svc.cluster.local",
+			envs:           runnerEnvs,
+			mount: []mount.Mount{
```

Comment:
Why don't you take this directly from the runner's container? Like what you did for gateway